extensions/api/data-management/transferprocess/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/transform/TransferProcessToTransferProcessDtoTransformerTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.transform;
+
+import org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.model.DataRequestDto;
+import org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.model.TransferProcessDto;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+class TransferProcessToTransferProcessDtoTransformerTest {
+    TransferProcessTransformerTestData data = new TransferProcessTransformerTestData();
+
+    TransferProcessToTransferProcessDtoTransformer transformer = new TransferProcessToTransferProcessDtoTransformer();
+    List<String> problems = new ArrayList<>();
+
+    @Test
+    void getInputType() {
+        assertThat(transformer.getInputType()).isEqualTo(TransferProcess.class);
+    }
+
+    @Test
+    void getOutputType() {
+        assertThat(transformer.getOutputType()).isEqualTo(TransferProcessDto.class);
+    }
+
+    @Test
+    void transform() {
+        assertThatEntityTransformsToDto();
+    }
+
+    @Test
+    void transform_whenInvalidState() {
+        var invalidStateCode = 0;
+        while (TransferProcessStates.from(invalidStateCode) != null) {
+            invalidStateCode++;
```

Comment:
it seems not relevant for the test how the invalid state is generated. Extracting it into a private method might make it easier to read:
```java
private int invalidStateCode() {
        var invalidStateCode = 0;
        while (TransferProcessStates.from(invalidStateCode) != null) {
            invalidStateCode++;
        }
        return invalidStateCode;
    }
```